src/diagnostics/MOM_obsolete_params.F90

```diff
@@ -68,6 +69,12 @@ subroutine find_obsolete_params(param_file)
        hint="Instead use OBC_SEGMENT_XXX_DATA.")
   call obsolete_char(param_file, "EXTEND_OBC_SEGMENTS", &
        hint="This option is no longer needed, nor supported.")
+  call read_param(param_file, "OBC_NUMBER_OF_SEGMENTS", nseg)
```

Comment:
set `nseg = 0` before calling read_param() so that it is initialized for the loop that follows.